test/bus_test.dart

```diff
@@ -269,6 +269,17 @@ void main() {
         );
       });
     });
+
+    group('invalid connection exceptions', () {
+      test('self-connecting logic', () {
+        try {
+          final a = Logic();
+          a <= a;
+        } on Exception catch (e) {
+          expect(e.runtimeType, InvalidConnectionException);
+        }
+      });
```

Comment:
We can add a test where `a <= b` and `b <= c` and `c <= a` with the `Wire` updates and it should be able to detect that too.